Fix infinite loop in test_new_tasks for some tasks

stephenroller opened the pull request:
test_new_tasks blindly iterates through all the tasks, which sometimes caused an infinite loop.

The problem was when we had imported a MultiTaskTeacher, which is done in some tasks. This causes a MultiTaskTeacher which tries to import MultiTaskTeacher, causing an infinite loop. This fix just makes sure MultiTaskTeacher isn't in the module directly, so that it isn't called by test_new_tasks. This isn't really a fix, so much as a bandaid so new copy-pasta doesn't have this problem.

This PR also makes it so that test_new_tasks checks all tasks and prints all errors, rather than just failing on the very first one. This can accelerate testing when multiple teachers are being introduced.

This PR does not pass test_new_tasks because some older tasks simply don't pass. I'm deferring fixing them for now.

hadasah commented:
lgtm, running test_new_tasks gives me all of the expected errors and exceptions on old tasks, but maybe kurt or emily want to get eyes on this too?

emilydinan commented on tests/datatests/test_new_tasks.py:
```diff
@@ -58,16 +60,23 @@ def test_verify_data(self):
+                        print('There are {} {} in {}.'.format(
                             log[key],
-                            log
+                            log,
```
i think you wanted this to be "key" here not log.

stephenroller commented:
Like I said, it doesn't pass because of old failures, but here's an example of the new error messages:

```
» python tests/datatests/test_new_tasks.py
Traceback (most recent call last):
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 731, in _create_task_agents
    create_agent = getattr(my_module, 'create_agents')
AttributeError: module 'parlai.tasks.babi.agents' has no attribute 'create_agents'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tests/datatests/test_new_tasks.py", line 65, in test_verify_data
    text, log = verify(opt, print_parser=False)
  File "/private/home/roller/working/parlai/parlai/scripts/verify_data.py", line 61, in verify
    world = create_task(opt, agent)
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 1029, in create_task
    world = create_task_world(opt, user_agents, default_world=default_world)
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 988, in create_task_world
    opt, user_agents, default_world=default_world
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 953, in _get_task_world
    task_agents = _create_task_agents(opt)
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 735, in _create_task_agents
    return create_task_agent_from_taskname(opt)
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 689, in create_task_agent_from_taskname
    task_agents = teacher_class(opt)
  File "/private/home/roller/working/parlai/parlai/tasks/babi/agents.py", line 64, in __init__
    self.task_num = task.split(':')[2]
IndexError: list index out of range
Got above exception in babi:Task10kTeacher
Traceback (most recent call last):
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 731, in _create_task_agents
    create_agent = getattr(my_module, 'create_agents')
AttributeError: module 'parlai.tasks.babi.agents' has no attribute 'create_agents'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tests/datatests/test_new_tasks.py", line 65, in test_verify_data
    text, log = verify(opt, print_parser=False)
  File "/private/home/roller/working/parlai/parlai/scripts/verify_data.py", line 61, in verify
    world = create_task(opt, agent)
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 1029, in create_task
    world = create_task_world(opt, user_agents, default_world=default_world)
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 988, in create_task_world
    opt, user_agents, default_world=default_world
  File "/private/home/roller/working/parlai/parlai/core/worlds.py", line 953, in _get_task_world
    task_agents = _create_task_agents(opt)
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 735, in _create_task_agents
    return create_task_agent_from_taskname(opt)
  File "/private/home/roller/working/parlai/parlai/core/agents.py", line 689, in create_task_agent_from_taskname
    task_agents = teacher_class(opt)
  File "/private/home/roller/working/parlai/parlai/tasks/babi/agents.py", line 46, in __init__
    self.task_num = task.split(':')[2]
IndexError: list index out of range
Got above exception in babi:Task1kTeacher
There are 30 empty_label_candidates in cbt:DefaultTeacher.
```
